src/lib/operators/aggregate_sort.cpp

```diff
@@ -443,29 +443,44 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
      * If the value clustering doesn't match the columns we group by, we need to re-sort the whole table
      * to achieve the value clustering we need.
      */
-    if (table_value_clustered_by && std::find(_groupby_column_ids.begin(), _groupby_column_ids.end(),
-                                              *(table_value_clustered_by->begin())) != _groupby_column_ids.end()) {
-      // Sort input table chunk_wise consecutively by the group by columns (stable sort)
-      sorted_table = _sort_within_chunks(input_table, _groupby_column_ids);
-    } else {
-      // sort input table in whole consecutively by the group by columns
-      for (const auto& column_id : _groupby_column_ids) {
-        auto table_wrapper = std::make_shared<TableWrapper>(sorted_table);
-        table_wrapper->execute();
-        auto sort =
-            std::make_shared<Sort>(table_wrapper, std::vector<SortColumnDefinition>{SortColumnDefinition{column_id}});
-        sort->execute();
-        sorted_table = sort->get_output();
+
+    bool is_grouped_by_all_value_clustered_segments = false;
```

Comment:
I don't understand what you are checking here. If a table is value clustered on two columns and only of them is part of the group-by list: why should that bother us?

+ const in loop, `auto` over `bool`.

Reply:
You are right. It is sufficient if the table is value clustered by at least one groupby column here